src/test/java/de/tum/in/www1/artemis/science/ScienceUtilService.java

```diff
@@ -0,0 +1,56 @@
+package de.tum.in.www1.artemis.science;
+
+import java.time.Duration;
+import java.time.ZonedDateTime;
+import java.util.Comparator;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.domain.science.ScienceEvent;
+import de.tum.in.www1.artemis.domain.science.ScienceEventType;
+import de.tum.in.www1.artemis.repository.science.ScienceEventRepository;
+
+@Service
+public class ScienceUtilService {
+
+    @Autowired
+    private ScienceEventRepository scienceEventRepository;
+
+    /**
+     * Creates a science event with the passed type and resource id.
+     *
+     * @param identity   The login of the user associated with the event.
+     * @param type       The type of the event.
+     * @param resourceId The id of the resource associated with the event.
+     */
+    public ScienceEvent createScienceEvent(String identity, ScienceEventType type, Long resourceId) {
+        ScienceEvent event = new ScienceEvent();
+        event.setIdentity(identity);
+        event.setTimestamp(ZonedDateTime.now());
+        event.setType(type);
+        event.setResourceId(resourceId);
+        return scienceEventRepository.save(event);
+    }
+
+    /**
+     * Comparator for comparing two science events.
+     * Allows for a 500ns difference between the timestamps due to the reimport from the csv export.
+     */
+    public static Comparator<ScienceEvent> scienceEventComparator = Comparator.comparing(ScienceEvent::getResourceId).thenComparing(ScienceEvent::getType)
+            .thenComparing((ScienceEvent e1, ScienceEvent e2) -> {
+
+                System.out.println(e1.getTimestamp());
+                System.out.println(e2.getTimestamp());
+
+                Duration d = Duration.between(e1.getTimestamp(), e2.getTimestamp());
+                System.out.println(d.getNano());
+                if (d.getNano() > 500 && d.getSeconds() >= 0) {
```

Comment:
I think this could be simplified by using the metod `d.toMillis()`.

Reply:
Working with millis is imo to broad of a timescale. During testing events are generated in a microsecond time frame. In case multiple scienceevents with the same type and resource occur in quick succession (wich i belive could also happen with real world data.) they would not be differentiated. Especially when rounding then leads to to big of a difference.